### PR TITLE
Upgrade rke2 to v1.21.3-rc1+rke2r2

### PR DESCRIPTION
**Pull request by galal-hussein**

### CODE REVIEW

File: channels-rke2.yaml [hunk 1]
```diff
@@ -35,6 +35,8 @@ releases:
         - rke2-ingress-nginx
         - rke2-kube-proxy
```
Comment (deniseschannon) [inline channels-rke2.yaml]: Should this be removed as an option if we are only allowing to enable the static pod?

Comment (galal-hussein) [reply channels-rke2.yaml]: hmmmm, okay that makes sense, removing

**Review by cwayne18 — APPROVED** [review]
LGTM

**Review by deniseschannon — APPROVED** [review]

**Comment by deniseschannon** [comment]
https://github.com/rancher/rancher/issues/33775
https://github.com/rancher/rancher/issues/32362
https://github.com/rancher/rancher/issues/33832
https://github.com/rancher/rancher/issues/33785

**Pull request merged**